src/EVM/Expr.hs

```diff
@@ -1214,29 +1214,29 @@ simplifyProp prop =
     go (PLEq a (Max _ b)) | a == b = PBool True
     go (PLEq (Sub a b) c) | a == c = PLEq b a
     go (PLT (Max (Lit a) b) (Lit c)) | a < c = PLT b (Lit c)
-    go (PLT (Lit 0) (Eq a b)) = mkPEq a b
+    go (PLT (Lit 0) (Eq a b)) = PEq a b
```

Comment:
I don't understand this... it's under `simplifyProp`, why not do `mkPEq` here? It's the most obvious thing to do I think? I must be not getting something. Under `simplify`, which simplifies `Expr`, we have:

```
    go (IsZero (IsZero (Eq x y))) = eq x y
```

Which is exactly the same idea -- `eq` will set the LHS, etc. So we seem to be inconsistent.

I don't mind it too much, but I just don't understand. Can you maybe explain what's the rationale and why is that `eq` ok, but this `mkPEq` not ok?

Reply:
I agree that here it would be good to simplify immediately. Do you agree as well, @zoep?

Reply:
I agree that for consistency it would be better to use `peq`/`mkPEq` here (though I understand that it doesn't matter so much as it will be simplified in the next iteration)